x-pack/plugins/ingest_manager/server/services/agents/acks.ts

```diff
@@ -92,8 +92,9 @@ function getLatestConfigIfUpdated(agent: Agent, actions: AgentAction[]) {
 
 function buildUpdateAgentConfig(agentId: string, config: FullAgentConfig) {
   const packages = config.inputs.reduce<string[]>((acc, input) => {
-    if (input.package && input.package.name && acc.indexOf(input.package.name) < 0) {
-      return [input.package.name, ...acc];
+    const packageName = input.meta?.package?.name;
+    if (packageName && acc.indexOf(packageName) < 0) {
+      return [packageName, ...acc];
```

Comment:
Do we need to handle cases where packages are removed?

Reply:
I rebuild this array each time from the full config, it should work for package removed too.

Reply:
May want to add a test case for this if possible